lib/src/yaml_edit/src/wrap.dart

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:collection' as collection;
+import 'package:collection/collection.dart';
+import 'package:source_span/source_span.dart';
+import 'package:yaml/yaml.dart';
+
+import 'equality.dart';
+import 'utils.dart';
+
+/// Returns a new [YamlList] constructed by applying [update] onto the [nodes]
+/// of this [YamlList].
+YamlList updatedYamlList(YamlList list, Function(List<YamlNode>) update) {
+  ArgumentError.checkNotNull(list, 'list');
+
+  final newNodes = [...list.nodes];
+  update(newNodes);
+  return wrapAsYamlNode(newNodes);
+}
+
+/// Returns a new [YamlMap] constructed by applying [update] onto the [nodes]
+/// of this [YamlMap].
+YamlMap updatedYamlMap(YamlMap map, Function(Map) update) {
+  ArgumentError.checkNotNull(map, 'map');
+
+  final dummyMap = deepEqualsMap();
+  dummyMap.addAll(map.nodes);
+
+  update(dummyMap);
+  final updatedMap = {};
+
+  /// This workaround is necessary since [yamlNodeFrom] will re-wrap
+  /// [YamlNode]s, so we need to unwrap them before passing them in.
```

Comment:
Is this still applicable now that we're using `wrapAsYamlNode`?

Reply:
nope - thanks for spotting!